azure/services/agentpools/client.go

```diff
@@ -50,19 +74,34 @@ func newAgentPoolsClient(subscriptionID string, baseURI string, authorizer autor
 
 // Get gets an agent pool.
 func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (result interface{}, err error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "agentpools.azureClient.Get")
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "agentpools.azureClient.Get")
 	defer done()
 
-	return ac.agentpools.Get(ctx, spec.ResourceGroupName(), spec.OwnerResourceName(), spec.ResourceName())
+	// Verify that we aren't currently rate limited or under active exponential backoff
```

Comment:
@CecileRobertMichon do you have any thoughts about any potential re-use opportunities that might prevent us from having to include this new rate limiting + backoff boilerplate into each service's get/update/delete flows?